src/events/IonEvent.ts

```diff
@@ -516,7 +516,7 @@ class IonFloatEvent extends AbstractIonEvent {
     }
     if (
       expected instanceof IonFloatEvent &&
-      this.ionValue === expected.ionValue
+      Object.is(this.ionValue, expected.ionValue)
```

Comment:
Now that we're using `Object#is`, [`NaN` will be considered equal to another `NaN`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is#Description). This means we can remove the special case `if` statement just above this one.